src/faldbt/project.py

```diff
@@ -449,12 +509,19 @@ def get_filtered_models(self, all, selected) -> List[DbtModel]:
                 "Cannot define models to run without selection flags or dbt run_results artifact"
             )
 
-        for node in self._get_models_with_keyword(self.keyword):
+        models = self._get_models_with_keyword(self.keyword)
+        if self._faldbt.method in ['test', 'build']:
+            tests = self._faldbt.list_tests()
+            models = self._map_tests_to_models(models, tests)
+
+        for node in models:
             if selected:
                 if node.unique_id in selected_ids:
                     filtered_models.append(node)
             elif all:
                 filtered_models.append(node)
+            elif self._faldbt.method in ['test', 'build'] and node.status == 'tested':
```

Comment:
This change and the one above makes me question if we are complicating things by having `fal run` handle both `dbt run` and `dbt test`.

I don't know if we should separate it as a different command.

Although the `dbt build` case is nice.

Reply:
I don't think we even need this case, since `node.status` is no longer `skipped` and the next condition will take care of it.